src/interactive/Toaster.jsx

```diff
@@ -0,0 +1,150 @@
+
+		this.timeouts = [];
+
+		this.cloneToast = this.cloneToast.bind(this);
+		this.dismissToast = this.dismissToast.bind(this);
+		this.clearTimeouts = this.clearTimeouts.bind(this);
+		this.restartTimeouts = this.restartTimeouts.bind(this);
+		this.setTimer = this.setTimer.bind(this);
+
+		this.state = {
+			toasts: this.props.toasts.map(this.cloneToast)
+		};
+	}
+
+	/**
+	 * @param {Object} dismissedToast - `Toast` component to remove from state
+	 * @returns undefined
+	 */
+	dismissToast(dismissedToast) {
+		this.setState({
+			toasts: this.state.toasts.filter(toast => dismissedToast.props.id !== toast.props.id)
+		});
+	}
+
+	/**
+	 * @param {Object} toast - `Toast` component to set a timeout on before passing to `dismissToast`
+	 * @returns undefined
+	 *
+	 * sets timers to delay a toast's dismissal
+	 */
+	setTimer(toast) {
+		this.timeouts.push(setTimeout(() => {
+			this.dismissToast(toast);
+		}, DELAY_TIME + (MARGINAL_DELAY * toast.key)));
+	}
+
+	/**
+	 * @returns undefined
+	 *
+	 * removes timers in order to prevent toasts from being automatically dismissed
+	 */
+	restartTimeouts() {
+		const toasts = this.state.toasts;
+
+		toasts.forEach((toast, i) => {
+			this.setTimer(toast);
+		});
+	}
+
+	/**
+	 * @returns undefined
+	 *
+	 * removes timers in order to prevent toasts from being automatically dismissed
+	 */
+	clearTimeouts() {
+		this.timeouts.forEach(clearTimeout);
+	}
+
+	componentWillUnmount() {
+		this.clearTimeouts();
+	}
+
+	componentDidMount() {
+		this.state.toasts && this.state.toasts.filter(t => t.props.autodismiss).map(this.setTimer);
+	}
+
+	componentWillReceiveProps(nextProps) {
+		const currentToasts = this.state.toasts;
+		const allToasts = currentToasts.concat(nextProps.toasts);
+
+		console.log('--------------------componentWillReceiveProps fired--------------------');
+
+		if (nextProps.toasts !== this.state.toasts) {
+			this.setState(() => ({ toasts: allToasts.map(this.cloneToast) }), () => {
+				this.state.toasts && this.state.toasts.filter(t => t.props.autodismiss).map(this.setTimer);
+			});
+		}
+	}
+
+	/**
+	 * @param {Object} Toast - `Toast` components to clone
+	 * @param {number} i - index of the `toast`
+	 * @returns {Array} `Toast` components with props from `Toaster`
+	 */
+	cloneToast(toast, i) {
+		const toastProps = {
+			key: i,
+			id: toast.props.id || `toast-${i}`,
+			dismissToast: this.dismissToast
+		};
+		return React.cloneElement(toast, toastProps);
+	}
+
+	render() {
+		const {
+			className,
+			toasts, // eslint-disable-line no-unused-vars
+			...other
+		} = this.props;
+
+		const classNames = cx(
+			'toaster',
+			className
+		);
+
+		return (
+			<div
+				className={classNames}
+				onMouseEnter={this.clearTimeouts}
+				onMouseLeave={this.restartTimeouts}
+				{...other}>
+				<TransitionGroup>
+					{
+						this.state.toasts.map((toast, i) => (
+							<CSSTransition
+								key={`csstrns-${i}`}
+								appear
+								timeout={250}
+								classNames="slide">
```

Comment:
can you put this closing `>` on a newline? A scan of this JSX makes it look like `{toast}` might be a prop